From baa1cb569a4b1faef80eefc9d4102c32fd8b0384 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:43:32 -0400
Subject: [ethernet] cxgb4: use dev_port to identify ports

Message-id: <1410475447-94392-74-git-send-email-svemuri@redhat.com>
Patchwork-id: 93838
O-Subject: [RHEL7.1 PATCH BZ 1124947 073/108] cxgb4: use dev_port to identify ports
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Commit 3f85944fe207d0225ef21a2c0951d4946fc9a95d ("net: Add sysfs file
for port number") introduce dev_port to network devices. cxgb4 adapters
have multiple ports on the same PCI function, and used dev_id to
identify those ports. That use was removed by commit
8c367fcbe6549195d2eb11e62bea233f811aad41 ("cxgb4: Do not set
net_device::dev_id to VI index"), since dev_id should be used only when
devices share the same MAC address.

Using dev_port for cxgb4 allows different ports on the same PCI function
to be identified.

Signed-off-by: Thadeu Lima de Souza Cascardo <cascardo@linux.vnet.ibm.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 40c9f8ab6c0f3e3c36dc3fb4ec81a25a0a207789)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c b/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
index d07c2ad..e3a7932 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
@@ -3963,6 +3963,7 @@ int t4_port_init(struct adapter *adap, int mbox, int pf, int vf)
   p->lport = j;
   p->rss_size = rss_size;
   memcpy(adap->port[i]->dev_addr, addr, ETH_ALEN);
+  adap->port[i]->dev_port = j;
 
   ret = ntohl(c.u.info.lstatus_to_modtype);
   p->mdio_addr = (ret & FW_PORT_CMD_MDIOCAP) ?
-- 
1.7.1